src/async_traits.rs

```
@@ -25,21 +28,32 @@ where
     async fn batch_execute_async(&self, query: &str) -> Result<(), DieselError>;
 }
 
+#[cfg(feature = "cockroach")]
+fn retryable_error(err: &DieselError) -> bool {
+    use diesel::result::DatabaseErrorKind::SerializationFailure;
+    match err {
+        DieselError::DatabaseError(SerializationFailure, boxed_error_information) => {
```

It looks like [`SerializationFailure` is defined to be synonymous with code 40001](https://docs.rs/diesel/latest/diesel/result/enum.DatabaseErrorKind.html#variant.SerializationFailure), which is the [condition that the CockroachDB docs say to look for](https://www.cockroachlabs.com/docs/v23.1/advanced-client-side-transaction-retries).  I think we shouldn't have to inspect the message?  (Not sure we _shouldn't_.  But I'm not sure what we _should_ do if we find a different message here.)

---

I'm happy to just ignore the error message, and rely on the error code. Not sure what to do if these are inconsistent -- I think that's more a question for "how much do we trust our database", because them being out-of-sync seems like undocumented territory.